plugins/cilium-cni/main.go

```diff
@@ -288,9 +288,16 @@ func prepareIP(ipAddr string, state *CmdState, mtu int) (*cniTypesV1.IPConfig, [
 	var (
 		routes []route.Route
 		gw     string
+		ip     netip.Addr
 	)
 
-	ip, err := netip.ParseAddr(ipAddr)
+	ipPrefix, err := netip.ParsePrefix(ipAddr)
+	if err != nil {
+		ip, err = netip.ParseAddr(ipAddr)
+	} else {
+		ip = ipPrefix.Addr()
+	}
```

Comment:
Could you add a small code comment explaining why this is necessary?

Reply:
Sure

Reply:
For context, I used the same logic as it was before: https://github.com/cilium/cilium/blob/c7516b9734a3c69db468dbacb34091347241cd47/pkg/addressing/ip.go#L102